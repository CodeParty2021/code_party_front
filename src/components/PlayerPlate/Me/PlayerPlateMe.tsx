import { Button } from "components/Button/Button";
import React from "react";
import { Badge } from "../components/Badge/Badge";
import { PlayerPlateCard } from "../components/PlayerPlateCard/PlayerPlateCard";
import { Status } from "../components/Status/Status";
import {
  ColumnBottomStye,
  ColumnMiddleStye,
  ColumnTopStye,
} from "./PlayerPlateMeStyle";

type Props = {
  userName?: string;
  userPhoto?: string;
  badge?: string;
  color?: "turquoise" | "leaf" | "orange" | "magenta";
  status: "default" | "ready";
  selectedCodeName?: string;
  onClickSelectCode?: () => void;
  onClickSelectOtherCode?: () => void;
};

export const PlayerPlateMe: React.FC<Props> = ({
  userName,
  badge,
  color,
  status,
  selectedCodeName,
  onClickSelectCode,
  onClickSelectOtherCode,
}) => {
  return (
    <PlayerPlateCard
      color={color}
      status={status}
      viewAlgo={true}
      size="L"
      playerIcon={<div>avatar</div>}
      columnTop={
        <ColumnTopStye>
          <span className="playerplateme-name">{userName}</span>
          <Badge badgeName={badge} color={color} />
        </ColumnTopStye>
      }
      columnMiddle={
        <ColumnMiddleStye>
          {status === "default" ? (
            <Button
              color="blue"
              size="L"
              value="コードを選ぶ"
              onClick={onClickSelectCode}
            />
          ) : (
            <>
              <Button
                color="white"
                size="M"
                value="別のコードを選ぶ"
                onClick={onClickSelectOtherCode}
              />
              <span className="playerplateme-codename">{selectedCodeName}</span>
            </>
          )}
        </ColumnMiddleStye>
      }
      columnBottom={
        <ColumnBottomStye>
          {status === "default" ? (
            <Status
              statusMessage="コード未選択"
              viewCheckMark={false}
              color={color}
            />
          ) : (
            <Status
              statusMessage="準備完了"
              viewCheckMark={true}
              color={color}
            />
          )}
        </ColumnBottomStye>
      }
    />
  );
};
