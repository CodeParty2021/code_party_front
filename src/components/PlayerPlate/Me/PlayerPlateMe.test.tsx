import React from "react";
import { PlayerPlateMe } from "./PlayerPlateMe";
import "jest-styled-components";
import { render } from "enzyme";

describe("<PlayerPlateMe />", () => {
  const defaultProps = {
    userName: "ほげ太郎",
    userPhoto: "/logo512.png",
    badge: "宇宙の探究者",
    color: "turquoise",
    status: "default",
    selectedCodeName: "ランダムに進むロボット.py",
  } as const;

  it("snapshot test", () => {
    const wrapper = render(<PlayerPlateMe {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("render test turquoise default", () => {
    expect(
      render(
        <PlayerPlateMe {...defaultProps} color="turquoise" status="default" />
      )
    ).toBeTruthy();
  });

  it("render test orange ready", () => {
    expect(
      render(<PlayerPlateMe {...defaultProps} color="orange" status="ready" />)
    ).toBeTruthy();
  });

  it("render test leaf", () => {
    expect(
      render(<PlayerPlateMe {...defaultProps} color="leaf" />)
    ).toBeTruthy();
  });

  it("render test magenta", () => {
    expect(
      render(<PlayerPlateMe {...defaultProps} color="magenta" />)
    ).toBeTruthy();
  });
});
