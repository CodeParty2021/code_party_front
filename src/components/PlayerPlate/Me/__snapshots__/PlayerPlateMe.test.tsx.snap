// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerPlateMe /> snapshot test 1`] = `
.c9 {
  display: block;
  width: 54px;
  height: 54px;
  padding: 1px;
  position: relative;
}

.c9 .avatar_main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2px;
  overflow: hidden;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #EAEBF1;
}

.c9 .avatar_main .avatar_img {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

.c9 .avatar_circle {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: 50%;
  border: 3px solid #FFFFFF;
}

.c9 .avatar_main {
  background-color: #ECFDFF;
}

.c9 .avatar_circle {
  border-color: #6CDBE5;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 700;
  line-height: 150%;
  color: #FFFFFF;
  background: #5E58C7;
  border-radius: 24px;
  padding-bottom: 8px;
  font-size: 24px;
}

.c7 .frame {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c7 .frame {
  background: linear-gradient(270deg,#4462FF 0%,#5F57FF 48.44%,#9670FF 100%);
}

.c7 .frame {
  border-radius: 24px;
  padding: 24px 64px;
  gap: 10px;
}

.c5 {
  width: 19px;
  height: 19px;
  fill: #34C5D5;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
}

.c4 .badge_name {
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
  color: #4A5373;
  font-weight: 700;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.c2 .algo_color_main {
  fill: #34C5D5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 32px;
  gap: 16px;
  position: relative;
  border-radius: 48px;
  background-color: #FAFAFA;
  width: 432px;
  height: 588px;
}

.c0 .playerplatecard_playericon {
  position: absolute;
  top: 30px;
  left: 30px;
}

.c0 .playerplatecard_window {
  position: relative;
  border-radius: 24px;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
}

.c0 .playerplatecard_window .c1 {
  position: absolute;
  bottom: -48px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0 .playerplatecard_column_top,
.c0 .playerplatecard_column_middle,
.c0 .playerplatecard_column_bottom {
  width: 100%;
}

.c0 .playerplatecard_column_top,
.c0 .playerplatecard_column_middle {
  border-bottom: 1px solid #D7DAE4;
}

.c0 .playerplatecard_border {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  border: 1px solid #D7DAE4;
  border-radius: 48px;
}

.c0 .playerplatecard_window {
  background-color: #A7ECF1;
}

.c0 .playerplatecard_border {
  border-color: #34C5D5;
}

.c0 .playerplatecard_border {
  border-color: #D7DAE4;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
}

.c8 .playerplate-status-message {
  font-weight: 700;
  font-size: 24px;
  line-height: 35px;
  color: #242A3D;
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 24px;
  gap: 4px;
}

.c3 .playerplateme-name {
  font-weight: 700;
  font-size: 30px;
  line-height: 43px;
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  padding-bottom: 16px;
  gap: 16px;
  width: 100%;
  height: 164px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c6 .playerplateme-codename {
  font-weight: 700;
  font-size: 18px;
  line-height: 150%;
  color: #4A5373;
  text-align: center;
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@supports not (gap:10px) {
  .c7 .frame > *:not(:first-child) {
    margin-top: 10px;
  }
}

@supports not (gap:16px) {
  .c4 > *:not(:first-child) {
    margin-left: 16px;
  }
}

@supports not (gap:16px) {
  .c0 > *:not(:first-child) {
    margin-top: 16px;
  }
}

@supports not (gap:16px) {
  .c8 > *:not(:first-child) {
    margin-left: 16px;
  }
}

@supports not (gap:4px) {
  .c3 > *:not(:first-child) {
    margin-top: 4px;
  }
}

@supports not (gap:16px) {
  .c6 > *:not(:first-child) {
    margin-top: 16px;
  }
}

<div
  class="c0"
  color="turquoise"
>
  <div
    class="playerplatecard_window"
  >
    <div
      class="c1 c2"
      color="turquoise"
    />
  </div>
  <div
    class="playerplatecard_column_top"
  >
    <div
      class="c3"
    >
      <span
        class="playerplateme-name"
      >
        ほげ太郎
      </span>
      <div
        class="c4"
      >
        <svg
          class="c5"
          color="turquoise"
          height="19px"
          width="19px"
          xlink="http://www.w3.org/1999/xlink"
          xmlns="http://www.w3.org/2000/svg"
        />
        <span
          class="badge_name"
        >
          宇宙の探究者
        </span>
        <svg
          class="c5"
          color="turquoise"
          height="19px"
          width="19px"
          xlink="http://www.w3.org/1999/xlink"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </div>
  </div>
  <div
    class="playerplatecard_column_middle"
  >
    <div
      class="c6"
    >
      <button
        class="c7"
        color="blue"
      >
        <div
          class="frame"
        >
          <span>
            コードを選ぶ
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    class="playerplatecard_column_bottom"
  >
    <div
      class=""
    >
      <div
        class="c8"
        color="turquoise"
      >
        <span
          class="playerplate-status-message"
        >
          コード未選択
        </span>
      </div>
    </div>
  </div>
  <div
    class="playerplatecard_playericon"
  >
    <div
      class="c9"
      color="turquoise"
      type="user"
    >
      <div
        class="avatar_main"
      >
        <img
          alt="ユーザアバター"
          class="avatar_img"
          src="/logo512.png"
        />
      </div>
      <div
        class="avatar_circle"
      />
    </div>
  </div>
  <div
    class="playerplatecard_border"
  />
</div>
`;
